### PR TITLE
boards: bl654_usb: use CDC ACM UART as serial backend

### LINKED ISSUE #39904: bl654_usb does not work with hci_usb sample application
**Describe the bug**
As noted in https://github.com/zephyrproject-rtos/zephyr/issues/38245 the hci_uart sample application does not work when building for bl654_usb due to the USB stack being used for console output

**Expected behavior**
For the sample application to work

### PR DESCRIPTION
**Pull request by jfischer-no**
usb: add option to initialize USB device support at boot

---

This patch adds new board specific Kconfig option
BL654_USB_SERIAL_BACKEND_CDCACM, which is enabled by
default except USB_DEVICE_BLUETOOTH class is used.
Applications that depend on usb_enable() or do not
require composite configuration cannot be used
with BL654 USB board without user intervention.

Fixes: #39904

### CODE REVIEW

File: subsys/usb/usb_device.c [hunk 1]
```diff
@@ -1628,7 +1630,14 @@ static int usb_device_init(const struct device *dev)
 
 	usb_set_config(device_descriptor);
 
+	if (IS_ENABLED(CONFIG_USB_DEVICE_INITIALIZE_AT_BOOT)) {
+		ret = usb_enable(NULL);
+		if (ret) {
```
Comment (carlescufi) [inline subsys/usb/usb_device.c:1635]: why do you need this `if` statement?

**Review by carlescufi — CHANGES_REQUESTED** [review]

File: subsys/usb/usb_device.c (hunk 1 above)
Comment (jfischer-no) [reply subsys/usb/usb_device.c:1635]: (⊙.☉)7, probably c&p

**Comment by lairdjm** [comment]
This doesn't solve the issue if users still need to supply a random Kconfig when building projects. https://github.com/zephyrproject-rtos/zephyr/pull/39987 fixes this without needing user intervention and providing an easy user experience

**Comment by jfischer-no** [comment]
> This doesn't solve the issue if users still need to supply a random Kconfig when building projects. #39987 fixes this without needing user intervention and providing an easy user experience

According to my understanding it is not possible and I have tried to explain it to you several times. Probably I am missing something, can you explain to me in what miraculous way it happens that option CONFIG_BOARD_ENABLE_USB_AUTOMATICALLY in #39987 is activated?

**Comment by lairdjm** [comment]
> > This doesn't solve the issue if users still need to supply a random Kconfig when building projects. #39987 fixes this without needing user intervention and providing an easy user experience
> 
> According to my understanding it is not possible and I have tried to explain it to you several times. Probably I am missing something, can you explain to me in what miraculous way it happens that option CONFIG_BOARD_ENABLE_USB_AUTOMATICALLY in #39987 is activated?

I built an application and it worked, so going off that which I also explained to you several times, yes, it is possible and works.
```
config USB_UART_CONSOLE
	default y if !USB_DEVICE_BLUETOOTH

config BOARD_ENABLE_USB_AUTOMATICALLY
	bool "Enable USB automatically"
	default y if USB_UART_CONSOLE
```

If USB_DEVICE_BLUETOOTH is not set, then USB_UART_CONSOLE is set, if this is set then BOARD_ENABLE_USB_AUTOMATICALLY is set which then enables USB automatically at startup with Kconfig specified in the build options via command line

**Comment by jfischer-no** [comment]
> ```
> config USB_UART_CONSOLE
> 	default y if !USB_DEVICE_BLUETOOTH
> 
> config BOARD_ENABLE_USB_AUTOMATICALLY
> 	bool "Enable USB automatically"
> 	default y if USB_UART_CONSOLE
> ```
> 
> If USB_DEVICE_BLUETOOTH is not set, then USB_UART_CONSOLE is set, if this is set then BOARD_ENABLE_USB_AUTOMATICALLY is set which then enables USB automatically at startup without any Kconfigs being specified in the build options via command line

There is no USB_UART_CONSOLE, but assuming it describes exact behavior desired by you, it is the same as:
 ```
 config BOARD_ENABLE_USB_AUTOMATICALLY
 	bool "Enable USB automatically"
 	default y if !USB_DEVICE_BLUETOOTH
 ```


or with 951633eeefbae31b06fa97484640b4c8c6cb3a34, 
```
config BL654_USB_SERIAL_BACKEND_CDCACM
	bool "Use CDC ACM UART as backend for BL654 USB adapter"
	default y if !USB_DEVICE_BLUETOOTH
	help
	  Use CDC ACM UART as backend for console or shell.

if BL654_USB_SERIAL_BACKEND_CDCACM

config USB_DEVICE_INITIALIZE_AT_BOOT
	default y

...

endif #BL654_USB_SERIAL_BACKEND_CDCACM
```

Is it correct so far? If yes please approve and accept the resulting disadvantages.

**Review by nordicjm — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**